surrogate_gossiper.go

```diff
@@ -40,10 +42,13 @@ func (*surrogateGossiper) Gossip() GossipData {
 // OnGossip should return "everything new I've just learnt".
 // surrogateGossiper doesn't understand the content of messages, but it can eliminate simple duplicates
 func (s *surrogateGossiper) OnGossip(update []byte) (GossipData, error) {
+	hash := fnv.New64a()
+	hash.Write(update)
+	updateHash := hash.Sum64()
```

Comment:
LGTM: given how the FNV hash function works, and that we compute the hash once and then re-use it in the loop, this should reduce run time complexity from `O(n^2.m)` to `O(n^2+m)` which is good for long messages.